rfc/graphql-schema/additionalTypeDefs/byPage/addresses.ts

```diff
@@ -0,0 +1,139 @@
+export default /* GraphQL */ `
+  type CompanyAddress {
+    entityId: Int!
+    firstName: String!
+    lastName: String!
+    address1: String!
+    address2: String
+    addressType: DestinationAddressType
+    city: String!
+    country: String
+    countryId: Int
+    countryCode: String!
+    stateOrProvince: String
+    stateId: Int
+    phone: String
+    postalCode: String
+    formFields: [CustomerFormFieldValue!]!
+
+    company: Company!
+
+    label: String!
+    isShipping: Boolean!
+    isBilling: Boolean!
+    isDefaultShipping: Boolean!
+    isDefaultBilling: Boolean!
+  }
+
+  type CompanyAddressEdge {
+    node: CompanyAddress!
+    cursor: String!
+  }
+
+  type CompanyAddressConnection {
+    edges: [CompanyAddressEdge!]!
+    pageInfo: PageInfo!
+    collectionInfo: CollectionInfo
+  }
+
+  input CompanyAddressFiltersInput {
+    search: String
+    city: String
+    state: String
+    country: String
+  }
+
+  extend type Company {
+    addresses(
+      filters: CompanyAddressFiltersInput
+      before: String
+      after: String
+      first: Int
+      last: Int
+    ): CompanyAddressConnection!
+  }
+
+  input CompanyAddressCreateInput {
+    firstName: String!
+    lastName: String!
+    address1: String!
+    address2: String
+    city: String!
+    countryCode: String!
+    stateOrProvince: String
+    phone: String
+    postalCode: String
+    extraFields: CustomerFormFieldsInput
+
+    label: String!
+    isShipping: Boolean!
+    isBilling: Boolean!
+    isDefaultShipping: Boolean!
+    isDefaultBilling: Boolean!
+  }
+
+  input CompanyAddressUpdateInput {
+    firstName: String
+    lastName: String
+    address1: String
+    address2: String
+    city: String
+    countryCode: String
+    stateOrProvince: String
+    phone: String
+    postalCode: String
+    extraFields: CustomerFormFieldsInput
+
+    label: String!
+    isShipping: Boolean
+    isBilling: Boolean
+    isDefaultShipping: Boolean
+    isDefaultBilling: Boolean
```

Comment:
would we want to consider seperate mutations for these more `meta` adjustments? 

`setAddressAsDefaultBilling`, `setAddressType` etc etc?

Reply:
there's a separate mutation for some of these adjustments, but there's a UI that allows for everything all at once, using separate mutation will effectively make it a "distributed transaction"